crmsh/bootstrap.py

```diff
@@ -828,32 +829,46 @@ def append_to_remote_file(fromfile, remote_node, tofile):
         utils.fatal("Failed to append contents of {} to {}:\n\"{}\"\n{}".format(fromfile, remote_node, err, err_details_string))
 
 
-def init_csync2():
+def configure_csync2():
+    """
+    Configure csync2, enable and start csync2.socket on both init and join side
+    """
     logger.info("Configuring csync2")
-    if os.path.exists(CSYNC2_KEY):
-        if not confirm("csync2 is already configured - overwrite?"):
-            return
-
-    invoke("rm", "-f", CSYNC2_KEY)
-    with logger_utils.status_long("Generating csync2 shared key (this may take a while)"):
+    me = utils.this_node()
+
+    # on join side
+    if _context.type == "join":
+        if not os.path.exists(CSYNC2_CFG):
+            utils.fatal("csync2 configure file {} not exists".format(CSYNC2_CFG))
+        if not re.search("host {};".format(me), CSYNC2_CFG):
+            cmd = "sed -i '/host\s*{};/a host {};' {}".format(_context.node_list[-1], me, CSYNC2_CFG)
+            utils.get_stdout_or_raise_error(cmd)
+            sync_file(CSYNC2_CFG)
+    # on init side
+    else:
+        if os.path.exists(CSYNC2_KEY):
+            if not confirm("csync2 is already configured - overwrite?"):
+                return
+        invoke("rm", "-f", CSYNC2_KEY)
         if not invokerc("csync2", "-k", CSYNC2_KEY):
             utils.fatal("Can't create csync2 key {}".format(CSYNC2_KEY))
 
-    csync2_file_list = ""
-    for f in FILES_TO_SYNC:
-        csync2_file_list += "include {};\n".format(f)
-
-    utils.str2file("""group ha_group
+        csync2_file_list = ""
+        for f in FILES_TO_SYNC:
+            csync2_file_list += "include {};\n".format(f)
+        utils.str2file("""group ha_group
 {{
 key /etc/csync2/key_hagroup;
 host {};
 {}
 }}
-    """.format(utils.this_node(), csync2_file_list), CSYNC2_CFG)
+        """.format(me, csync2_file_list), CSYNC2_CFG)
 
+    logger.info("Enable and start csync2.socket")
     utils.start_service("csync2.socket", enable=True)
-    with logger_utils.status_long("csync2 checking files"):
-        invoke("csync2", "-cr", "/")
+
+    if _context.type == "join":
+        logger.warning("To initially synchronize all files configured by csync2, please run \"crm cluster init csync2\" on suitable time(this may take a while)")
```

Comment:
```suggestion
        logger.warning("csync2 database is not initiated yet. Before using csync2 for the first time, please run \"crm cluster init csync2\" on any one node. Note, this may take a while.")
```